pkg/settings/setting.go

```diff
@@ -256,3 +257,28 @@ func GetSettingByID(id string) string {
 	}
 	return provider.Get(id)
 }
+
+func MergeSettingsWithAgentEnvVars(envVars []v1.EnvVar) {
```

Comment:
This won't update the slice passed. Proof: https://play.golang.org/p/4OD4P4u4Sve

It may be a moot point though. Do we actually want `cluster.Spec.AgentEnvVars` updated to now include these select Rancher settings or do we just need a combined list to use further in the call sequence? From what I see it's the latter, so it makes more sense for the function to return a slice.

Alternatively, you can have func like `GetSettingsAsAgentEnvVars` taking no arguments and do appending on the caller's side. It may actually be cleaner.

Reply:
I really have to get out of the whole "arrays are pointers" thing. I do think this is a good topic for discussion though, we need to be storing them somewhere (preferably relating to the agent) in order to detect if a redeploy is necessary, but if it's the `AgentEnvVars` then we should think about whether or not we want to merge. We can just prepend the environment variables to `AgentEnvVars`, but this affects how they show up in "Edit Config/YAML".

Reply:
Yup. Also, thinking more about it the second suggestion (to have func that returns settings as env variables) may not make sense as then merging logic will move down to the caller. Not sure. But yeah, we need to discuss what you mentioned above.